core/shared/src/main/scala/spire/math/poly/MultivariatePolynomial.scala

```diff
@@ -0,0 +1,214 @@
+package spire.math.poly
+
+import scala.annotation.tailrec
+import scala.collection.mutable.ArrayBuilder
+import scala.collection.mutable.Set
+import scala.reflect._
+import scala.{specialized => spec}
+import spire.algebra._
+import spire.implicits._
+import spire.math._
+
+
+class MultivariatePolynomial[@spec(Double) C: Order] private[spire] (val terms: Array[Monomial[C]])
```

Comment:
The `terms` mutable array is public; considering the rest of the code is written without much emphasis on performance, we can replace it by a `Seq`, and forget about the `ClassTag`.

Same remark as above regarding the `Order` requirement: this forbids the use of complex coefficients.